### PR TITLE
python3Packages.rstcheck: init at 3.3.1

### PR DESCRIPTION
**Pull request by dyrnade**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -1515,6 +1515,12 @@
     githubId = 977929;
     name = "Cody Allen";
   };
+  cemguresci = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:1518]: Please put this into a separate commit with the message ``maintainers: add cemguresci``.

File: pkgs/development/python-modules/rstcheck/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = "https://github.com/myint/rstcheck";
+    description = "Checks syntax of reStructuredText and code blocks nested within it";
+    license = with licenses; [ mit ];
+    maintainers = with maintainers; [ cemguresci ];
+  };
+}
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rstcheck/default.nix:22]: Missing final new line.

File: pkgs/development/python-modules/rstcheck/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  version = "3.3.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "07vl2p16fw0jayfcn424ixfc3cyhj8pnm89b83h70hm5as9ggi4j";
+  };
+
+  propagatedBuildInputs = [ docutils ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rstcheck/default.nix:14]: Please add a comment why tests are disabled.

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -6794,6 +6794,8 @@ in {
   else
     callPackage ../development/python-modules/rsa/4_0.nix { };
 
+  rstcheck = callPackage ../development/python-modules/rstcheck { buildPythonPackage = pkgs.python3Packages.buildPythonPackage; inherit (pkgs) docutils; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:6797]:
```suggestion
  rstcheck = callPackage ../development/python-modules/rstcheck {
    inherit (pkgs) docutils;
  };
```

If you want to limit it to python3 please use my above suggestion.

File: pkgs/development/python-modules/rstcheck/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{ lib, buildPythonPackage, fetchPypi, docutils }:
+
+buildPythonPackage rec {
+  pname = "rstcheck";
+  version = "3.3.1";
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rstcheck/default.nix:6]:
```suggestion

  disabled = isPy27;

```

and add isPy27 to inputs.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Review by jonringer — COMMENTED** [review]
To comply with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes) please have the commit message name be of the format
```
<pkg-name>: <subject-line>
```
for more examples, please look at https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

in your case, the commit message should be:
```
python3Packages.rstcheck: init at 3.3.1
```

File: pkgs/development/python-modules/rstcheck/default.nix (hunk 5 above)
Comment (dyrnade) [reply pkgs/development/python-modules/rstcheck/default.nix:6]: Thanks, i didnt know this.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (dyrnade) [reply maintainers/maintainer-list.nix:1518]: Added.

File: pkgs/top-level/python-packages.nix [hunk 6]
```diff
@@ -6794,6 +6794,8 @@ in {
   else
     callPackage ../development/python-modules/rsa/4_0.nix { };
 
+  rstcheck = callPackage ../development/python-modules/rstcheck { inherit (pkgs) docutils; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:6797]:
```suggestion
  rstcheck = callPackage ../development/python-modules/rstcheck {
    inherit (pkgs) docutils;
  };
```

File: pkgs/development/python-modules/rstcheck/default.nix [hunk 7]
```diff
@@ -0,0 +1,25 @@
+{ lib, buildPythonPackage, fetchPypi, docutils, isPy27 }:
+
+buildPythonPackage rec {
+  pname = "rstcheck";
+  version = "3.3.1";
+
+  disabled = isPy27;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rstcheck/default.nix:7]:
```suggestion
  version = "3.3.1";
  disabled = isPy27;
```

File: pkgs/development/python-modules/rstcheck/default.nix [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "07vl2p16fw0jayfcn424ixfc3cyhj8pnm89b83h70hm5as9ggi4j";
+  };
+
+  propagatedBuildInputs = [ docutils ];
+
+  # project has no tests
+  doCheck = true;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rstcheck/default.nix:17]:
```suggestion
  doCheck = true;
  
  pythonImportsCheck = [ "rstcheck" ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110553` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.rstcheck: log <a href="https://termbin.com/5fcn">https://termbin.com/5fcn</a></li>
<li>python39Packages.rstcheck: log <a href="https://termbin.com/8ici">https://termbin.com/8ici</a></li>
  </ul>
</details>
<details>
<summary>
1 package built:
</summary>
<ul>
<li>python38Packages.rstcheck</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python38Packages.rstcheck:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110553` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.rstcheck: log <a href="https://termbin.com/dohc">https://termbin.com/dohc</a></li>
<li>python39Packages.rstcheck: log <a href="https://termbin.com/0yxl">https://termbin.com/0yxl</a></li>
  </ul>
</details>
<details>
<summary>
1 package built:
</summary>
<ul>
<li>python38Packages.rstcheck</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python38Packages.rstcheck:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>

File: maintainers/maintainer-list.nix [hunk 9]
```diff
@@ -10484,4 +10490,5 @@
     github = "zupo";
     githubId = 311580;
   };
+
 }
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:10494]:
```suggestion
  };
}
```

**Comment by fabaff** [comment]
Superseded by #136209
